provider/devices/common.go

```diff
@@ -908,3 +931,68 @@ func checkAppiumUp(device *models.Device) {
 		loops++
 	}
 }
+
+func updateDeviceWithGlobalSettings(dbDevice *models.Device) error {
+	globalSettings, err := db.GetGlobalStreamSettings()
+	if err != nil {
+		return fmt.Errorf("failed to get global stream settings: %v", err)
+	}
+
+	dbDevice.StreamTargetFPS = globalSettings.TargetFPS
+	dbDevice.StreamJpegQuality = globalSettings.JpegQuality
+
+	// Check the device OS before assigning the scaling factor
+	if dbDevice.OS == "android" {
+		dbDevice.StreamScalingFactor = globalSettings.ScalingFactorAndroid
+	} else if dbDevice.OS == "ios" {
+		dbDevice.StreamScalingFactor = globalSettings.ScalingFactoriOS
+	}
+
+	return nil
+}
+
+func UpdateDevicesStreamSettings() {
+	ticker := time.NewTicker(1 * time.Second)
+	defer ticker.Stop()
+
+	for range ticker.C {
+		for _, dbDevice := range DBDeviceMap {
+			time.Sleep(1 * time.Second)
```

Comment:
I think this sleep should be outside the DBDeviceMap loop, why are we waiting per device instead of per ticker?

Reply:
Really... this sleep is unnecessary inside the loop since we're already waiting 1 second using the ticker. The fix is available in the [commit](https://github.com/shamanec/GADS/pull/145/commits/039fc1a554e9e13d4761a87cffdbac16725f4e86)

Reply:
yeah, we're both stoopid :D